Update storage and transfer size units to SI standards (#33)
raspberrypi/style-guide: style-guide.md

 
###B 

> *   b - bit = the most basic unit of information in computing and digital communication. The name is a portmanteau of **binary** and **digit**! Always use a space between the number and the unit, preferably nonbreaking. _Use "b" as a suffix only, otherwise explicitly write out "bit"._

Reviewer: Might also be worth pointing out the way that we hyphenate "bit" when we use it an an adjectival sense? e.g. "These 3 bits control the foo" or "The xyzyz field is 6 bits wide" vs. "BAR is a 32-bit register" or "AArch64 is a 64-bit CPU architecture".

Author: ```suggestion
*   b - bit = the most basic unit of information in computing and digital communication. The name is a portmanteau of **binary** and **digit**! Always use a space between the number and the unit, preferably nonbreaking. _Use "b" as a suffix only, otherwise explicitly write out "bit"._ When used in an adjectival sense (e.g. 'The Sega Genesis is a 16-bit home video game console'), hyphenate. When counting something (e.g. 'These 3 bits control the foo'), do not hyphenate. 
```

*   b/s - bits per second. Always use a space between the number and the unit, preferably nonbreaking.

> *   B - byte = 2<sup>3</sup> bit = 8 bit

Reviewer: I'm not sure if the "2<sup>3</sup>" is really relevant / useful here? :man_shrugging:

Author: True, but I figured it doesn't hurt to be painfully explicit 😉

> *   B - byte = 2<sup>3</sup> bit = 8 bit

Reviewer 2: Aside: if you're mentioning the etymology of `bit`, maybe do the same for `byte` = `binary term`, rather than `bite` which could get confused with `bit`. See https://www.cs.cmu.edu/~fgandon/documents/lecture/uk1999/binary/HandOut.pdf and https://en.wikipedia.org/wiki/Byte 😄

> *   B/s - bytes per second. Always use a space between the number and the unit, preferably nonbreaking.

Author: ```suggestion
*   B/s - bytes per second. Always use a space between the number and the unit, preferably nonbreaking.
*   byte - 8 bits. The name is both a portmanteau of **binary** and **term** as well as a deliberate respelling of bite to avoid accidental mutation to bit. Always use a space between the number and the unit, preferably nonbreaking. _Use "B" as a suffix only, otherwise explicitly write out "byte"._ When used as a compound adjective, (e.g. 'The key is stored in a 2-byte register'), hyphenate. When used as a quantifier (e.g. 'These 3 bytes control the foobarbaz'), do not hyphenate. 
```

 
###B 

> *   b - bit = the most basic unit of information in computing and digital communication. The name is a portmanteau of **binary** and **digit**! Always use a space between the number and the unit, preferably nonbreaking. _Use "b" as a suffix only, otherwise explicitly write out "bit"._ When used as a compound adjective, (e.g. 'The Sega Genesis is a 16-bit home video game console'), hyphenate. When used as a quantifier, (e.g. 'These 3 bits control the foo'), do not hyphenate. 

Reviewer: \*cough\*  Megadrive :stuck_out_tongue_winking_eye:

Author: Oh _man_. TIL! And here I thought UK folks were just funny for adding extra Us and hating Z. We clearly missed a fantastic opportunity to name the M.2 HAT+ the "gigadrive".

> *   b - bit = the most basic unit of information in computing and digital communication. The name is a portmanteau of **binary** and **digit**! Always use a space between the number and the unit, preferably non-breaking. _Use "b" as a suffix only, otherwise explicitly write out "bit"._ When used as a compound adjective, (e.g. 'The Sega Genesis is a 16-bit home video game console'), hyphenate. When used as a quantifier, (e.g. 'These 3 bits control the foo'), do not hyphenate. 

Author: ```suggestion
*   b - bit = the most basic unit of information in computing and digital communication. The name is a portmanteau of **binary** and **digit**! Always use a space between the number and the unit, preferably non-breaking. _Use "b" as a suffix only, otherwise explicitly write out "bit"._ When used as a compound adjective, (e.g. 'The Mega Drive is a 16-bit home video game console'), hyphenate. When used as a quantifier, (e.g. 'These 3 bits control the foo'), do not hyphenate. 
```

*   b/s - bits per second. Always use a space between the number and the unit, preferably nonbreaking.

> *   B - byte = 2<sup>3</sup> bit = 8 bit

Author: ```suggestion
*   B - byte = 2<sup>3</sup> b = 8 b
```

*   B/s - bytes per second. Always use a space between the number and the unit, preferably nonbreaking.

> *   byte - 8 bits. The name is both a portmanteau of **binary** and **term** as well as a deliberate respelling of bite to avoid accidental mutation to bit. Always use a space between the number and the unit, preferably nonbreaking. _Use "B" as a suffix only, otherwise explicitly write out "byte"._ When used as a compound adjective, (e.g. 'The key is stored in a 2-byte register'), hyphenate. When used as a quantifier (e.g. 'These 3 bytes control the foobarbaz'), do not hyphenate. 

Author: ```suggestion
*   byte - 8 b. The name is both a portmanteau of **binary** and **term** as well as a deliberate respelling of bite to avoid accidental mutation to bit. Always use a space between the number and the unit, preferably nonbreaking. _Use "B" as a suffix only, otherwise explicitly write out "byte"._ When used as a compound adjective, (e.g. 'The key is stored in a 2-byte register'), hyphenate. When used as a quantifier (e.g. 'These 3 bytes control the foobarbaz'), do not hyphenate. 
```

*   g – gram(s). Use a numeral with g and no space between the numeral and g. *The teabag weighs 7g.* 
*   gamer
*   games console. Not "game console", which is the US usage: British usage is "games". See also maths/math.

> *   Gib - gibibit =  2<sup>30</sup> bit = 1073741824 bit. Always use a space between the number and the unit, preferably nonbreaking.

Reviewer: "2<sup>30</sup> bit = 1073741824 bit" - both "bit" here should probably be pluralised as "bits" ?

Author: ```suggestion
*   Gib - gibibit =  2<sup>30</sup> b = 1073741824 b. Always use a space between the number and the unit, preferably nonbreaking.
```
Technically, since this is a suffix unit, we should follow the style guide above and use "b" 😜

*   Gib/s - gibibits per second. Always use a space between the number and the unit, preferably nonbreaking.

> *   Gb - gigabit = 10<sup>9</sup> bit = 1000000000 bit. Always use a space between the number and the unit, preferably nonbreaking.

Author: ```suggestion
*   Gb - gigabit = 10<sup>9</sup> b = 1000000000 b. Always use a space between the number and the unit, preferably nonbreaking.
```

*   Gb/s - gigabits per second. Always use a space between the number and the unit, preferably nonbreaking.

> *   GiB - gibibyte =  2<sup>30</sup> B = 1073741824 B. Always use a space between the number and the unit, preferably nonbreaking.

Reviewer: Does this mean that our documentation ought to refer to e.g. "32 GiB SD card" even though they're **always** labelled as "32GB" ? :thinking: 
https://www.amazon.co.uk/SanDisk-Ultra-Memory-Class-UHS-I/dp/B08GYG6T12/

Author: ...yes! Better to be explicit, though for introductory sections we may want to clarify that "GB" is sorta kinda equivalent.

Reviewer 3: Because we're also using the style guide for things like blog posts, where we'd want to say "32GB", please can we reinstate an entry for GB? With something like "what is popularly referred to as a '32GB SD card' is, strictly speaking, a '32 GiB SD card'. Use the latter in copy aimed at a technical audience (eg datasheets), but use the former in copy intended for a general audience, such as blog posts and press releases".

Reviewer: Does the style guide also apply to e.g. https://www.raspberrypi.com/products/raspberry-pi-5/ ?

Reviewer 3: Yes - I'd call that "copy intended for a general audience".

> *   GiB - gibibyte =  2<sup>30</sup> B = 1073741824 B. Always use a space between the number and the unit, preferably nonbreaking.

Author: ```suggestion
*   GiB - gibibyte =  2<sup>30</sup> B = 1073741824 B. Always use a space between the number and the unit, preferably nonbreaking. Preferred for copy aimed at a technical audience, like datasheets and reference documentation. For blog posts, use GB instead.
```

*   GiB/s - gibibytes per second. Always use a space between the number and the unit, preferably nonbreaking.

> *   GB - gigabyte = 10<sup>9</sup> B = 1000000000 B. Always use a space between the number and the unit, preferably nonbreaking.

Author: ```suggestion
*   GB - gigabyte = 10<sup>9</sup> B = 1000000000 B. Always use a space between the number and the unit, preferably nonbreaking. Preferred for content aimed at a general audience, like blog posts. For copy aimed at a technical audience, like datasheets and reference documentation, use GB instead to avoid looking like a pedantic nerd.
```

*   g – gram(s). Use a numeral with g and no space between the numeral and g. *The teabag weighs 7g.* 
*   gamer
*   games console. Not "game console", which is the US usage: British usage is "games". See also maths/math.
*   Gib - gibibit =  2<sup>30</sup> b = 1073741824 b. Always use a space between the number and the unit, preferably nonbreaking.
*   Gib/s - gibibits per second. Always use a space between the number and the unit, preferably nonbreaking.
*   Gb - gigabit = 10<sup>9</sup> b = 1000000000 b. Always use a space between the number and the unit, preferably nonbreaking.
*   Gb/s - gigabits per second. Always use a space between the number and the unit, preferably nonbreaking.
*   GiB - gibibyte =  2<sup>30</sup> B = 1073741824 B. Always use a space between the number and the unit, preferably nonbreaking. Preferred for copy aimed at a technical audience, like datasheets and reference documentation. For content aimed at a general audience, like blog posts, use GB instead to avoid looking like a pedantic nerd.
*   GiB/s - gibibytes per second. Always use a space between the number and the unit, preferably nonbreaking.

> *   GB - gigabyte = 10<sup>9</sup> B = 1000000000 B. Always use a space between the number and the unit, preferably nonbreaking. Preferred for content aimed at a general audience, like blog posts. For copy aimed at a technical audience, like datasheets and reference documentation, use GB instead to avoid looking like a pedantic nerd.

Author: ```suggestion
*   GB - gigabyte = 10<sup>9</sup> B = 1000000000 B. Always use a space between the number and the unit, preferably nonbreaking. Preferred for content aimed at a general audience, like blog posts. For copy aimed at a technical audience, like datasheets and reference documentation, use GiB instead for pedantic nerd cred.
```

 
###K 

*   k – _do not use_ as an abbreviation for thousand. When talking about kilobytes, use kB, not k, which is outdated. *The original BBC Micro had 32kB memory.*
*   Kib - kibibit = 2<sup>10</sup> bit = 1024 bit
*   Kib/s - kibibits per second. Always use a space between the number and the unit, preferably nonbreaking.
*   kb - kilobit = 10<sup>3</sup> bit = 1000 bit. Always use a space between the number and the unit, preferably nonbreaking.
*   kb/s – kilobits per second. Always use a space between the number and the unit, preferably nonbreaking.

> *   kB – kilobyte. 1000 B. _Rarely used; you probably want KiB._ Always use a space between the number and the unit, preferably nonbreaking.
> *   kB/s – kilobytes per second. _Rarely used; you probably want KiB/s._ Always use a space between the number and the unit, preferably nonbreaking.
> *   KiB - kibibyte = 2<sup>10</sup> B = 1024 B. Always use a space between the number and the unit, preferably nonbreaking.

Author: ```suggestion
*   kB – kilobyte. 1000 B. _Rarely used; you probably want KiB._ Always use a space between the number and the unit, preferably nonbreaking.
*   KB - _do not use_. Use kB or KiB instead.
*   KB/s - _do not use_. Use kB/s or KiB/s instead.
*   kB/s – kilobytes per second. _Rarely used; you probably want KiB/s._ Always use a space between the number and the unit, preferably nonbreaking.
*   KiB - kibibyte = 2<sup>10</sup> B = 1024 B. Always use a space between the number and the unit, preferably nonbreaking.
```

 
###K 

> *   k – _do not use_ as an abbreviation for thousand. When talking about kilobytes, use kB, not k, which is outdated. *The original BBC Micro had 32kB memory.*

Reviewer: In the example here, you seem to be breaking your "Always use a space between the number and the unit" rule? :stuck_out_tongue_winking_eye:  Also, did you mean "32 KiB"?

Author: ```suggestion
*   k – _do not use_ as an abbreviation for thousand. When talking about kilobytes, use kB, not k, which is outdated. *The original BBC Micro had 32 kiB memory.*
```
Right you are!

> *   k – _do not use_ as an abbreviation for thousand. When talking about kilobytes, use kB, not k, which is outdated. *The original BBC Micro had 32 kiB memory.*

Author: ```suggestion
*   k – _do not use_ as an abbreviation for thousand. When talking about kilobytes, use kB, not k, which is outdated. *The original BBC Micro had 32 KiB memory.*
```

LOL. Growing pains.

> *   Kib - kibibit = 2<sup>10</sup> bit = 1024 bit
> *   Kib/s - kibibits per second. Always use a space between the number and the unit, preferably nonbreaking.
> *   kb - kilobit = 10<sup>3</sup> bit = 1000 bit. Always use a space between the number and the unit, preferably nonbreaking.

Author: ```suggestion
*   Kib - kibibit = 2<sup>10</sup> b = 1024 b
*   Kib/s - kibibits per second. Always use a space between the number and the unit, preferably nonbreaking.
*   kb - kilobit = 10<sup>3</sup> b = 1000 b. Always use a space between the number and the unit, preferably nonbreaking.
```

*   kb/s – kilobits per second. Always use a space between the number and the unit, preferably nonbreaking.
*   kB – kilobyte. 1000 B. _Rarely used; you probably want KiB._ Always use a space between the number and the unit, preferably nonbreaking. Always use a _lowercase_ k.

> *   kB/s – kilobytes per second. _Rarely used; you probably want KiB/s._ Always use a space between the number and the unit, preferably nonbreaking. Always use a _lowercase_ k.

Author: ```suggestion
*   kB/s – kilobytes per second. Always use a space between the number and the unit, preferably nonbreaking. Always use a _lowercase_ k.
```

> *   kB – kilobyte. 1000 B. _Rarely used; you probably want KiB._ Always use a space between the number and the unit, preferably nonbreaking.
> *   KB - _do not use_. Use kB or KiB instead.
> *   KB/s - _do not use_. Use kB/s or KiB/s instead.
> *   kB/s – kilobytes per second. _Rarely used; you probably want KiB/s._ Always use a space between the number and the unit, preferably nonbreaking.

Author: ```suggestion
*   kB – kilobyte. 1000 B. _Rarely used; you probably want KiB._ Always use a space between the number and the unit, preferably nonbreaking. Always use a _lowercase_ k.
*   kB/s – kilobytes per second. _Rarely used; you probably want KiB/s._ Always use a space between the number and the unit, preferably nonbreaking. Always use a _lowercase_ k.
*   KB - _do not use_. Use kB or KiB instead.
*   KB/s - _do not use_. Use kB/s or KiB/s instead.
```

> *   KiB - kibibyte = 2<sup>10</sup> B = 1024 B. Always use a space between the number and the unit, preferably nonbreaking.

Reviewer: So, always a lower-case k when talking about kilobytes (kB), but always an upper-case K when talking about kibibytes (KiB) ? That's probably a distinction that is worth pointing out explicitly :thinking:

Author: Agreed. Updated here: https://github.com/raspberrypi/style-guide/pull/33/files#r1677747454

*   Master Boot Record
*   Mastodon – social network. Please don't use their *Toot* coinage: use *post* instead.
*   Maths, not math. Mathematics is also acceptable.
*   Mbit – megabit (not Mb, as this is too easily confused with MB)

> *   MB – megabyte (not MByte). No space between the number and MB. *My old Amiga only had 130MB 

Reviewer: Might be worth retaining this example-usage?

*   Mib - mebibit = 2<sup>20</sup> b = 1048576 b. Always use a space between the number and the unit, preferably nonbreaking.
*   Mib/s - mebibits per second. _Rarely used; you probably want Mb/s._ Always use a space between the number and the unit, preferably nonbreaking.

> *   MB - megabyte = 10<sup>6</sup> B = 1000000 B. _Rarely used for storage; you probably want MiB._ Always use a space between the number and the unit, preferably nonbreaking.

Reviewer: "Rarely used for storage" - IIRC mechanical hard-drives tend to use the base-10 notation (Gigabytes), but flash devices (SD-cards, NVMe SSDs etc.) tend to use the base-2 notation (Gibibytes) ?

Author: ```suggestion
*   MB - megabyte = 10<sup>6</sup> B = 1000000 B. Always use a space between the number and the unit, preferably nonbreaking. *My old Amiga only had 130 MB drive space.*
```

*   Mib/s - mebibits per second. _Rarely used; you probably want Mb/s._ Always use a space between the number and the unit, preferably nonbreaking.
*   MiB - mebibyte = 2<sup>20</sup> B = 1048576 B. Always use a space between the number and the unit, preferably nonbreaking.

> *   MiB/s - mebibytes per second. _Rarely used; you probably want MB/s._ Always use a space between the number and the unit, preferably nonbreaking.

Reviewer: I can't tell why you've got both "kB/s – kilobytes per second. _Rarely used; you probably want KiB/s._" and "MiB/s - mebibytes per second. _Rarely used; you probably want MB/s._" ? :thinking: :face_with_spiral_eyes:

> *   Mb - megabit. Always use a space between the number and the unit, preferably nonbreaking.
> *   Mb/s = megabits per second. Always use a space between the number and the unit, preferably nonbreaking.
> *   MB - megabyte = 10<sup>6</sup> B = 1000000 B. _Rarely used for storage; you probably want MiB._ Always use a space between the number and the unit, preferably nonbreaking.
> *   MB/s - megabytes per second. Always use a space between the number and the unit, preferably nonbreaking.
> *   Mib - mebibit = 2<sup>20</sup> bit = 1048576 bit. Always use a space between the number and the unit, preferably nonbreaking.
> *   Mib/s - mebibits per second. _Rarely used; you probably want Mb/s._ Always use a space between the number and the unit, preferably nonbreaking.
> *   MiB - mebibyte = 2<sup>20</sup> B = 1048576 B. Always use a space between the number and the unit, preferably nonbreaking.
> *   MiB/s - mebibytes per second. _Rarely used; you probably want MB/s._ Always use a space between the number and the unit, preferably nonbreaking.

Author: ```suggestion
*   Mb - megabit = 1000000 b. Always use a space between the number and the unit, preferably nonbreaking.
*   Mb/s = megabits per second. Always use a space between the number and the unit, preferably nonbreaking.
*   Mib - mebibit = 2<sup>20</sup> b = 1048576 b. Always use a space between the number and the unit, preferably nonbreaking.
*   Mib/s - mebibits per second. _Rarely used; you probably want Mb/s._ Always use a space between the number and the unit, preferably nonbreaking.
*   MB - megabyte = 10<sup>6</sup> B = 1000000 B. _Rarely used for storage; you probably want MiB._ Always use a space between the number and the unit, preferably nonbreaking.
*   MB/s - megabytes per second. Always use a space between the number and the unit, preferably nonbreaking.
*   MiB - mebibyte = 2<sup>20</sup> B = 1048576 B. Always use a space between the number and the unit, preferably nonbreaking.
*   MiB/s - mebibytes per second. _Rarely used; you probably want MB/s._ Always use a space between the number and the unit, preferably nonbreaking.
```
